routers/web/auth/oauth.go

```diff
@@ -1275,8 +1281,12 @@ func oAuth2UserLoginCallback(ctx *context.Context, authSource *auth.Source, requ
 		if err.Error() == "securecookie: the value is too long" || strings.Contains(err.Error(), "Data too long") {
 			log.Error("OAuth2 Provider %s returned too long a token. Current max: %d. Either increase the [OAuth2] MAX_TOKEN_LENGTH or reduce the information returned from the OAuth2 provider", authSource.Name, setting.OAuth2.MaxTokenLength)
 			err = fmt.Errorf("OAuth2 Provider %s returned too long a token. Current max: %d. Either increase the [OAuth2] MAX_TOKEN_LENGTH or reduce the information returned from the OAuth2 provider", authSource.Name, setting.OAuth2.MaxTokenLength)
+			return nil, goth.User{}, err
 		}
-		return nil, goth.User{}, err
+
+		log.Error("OAuth2 Provider %s error(start BeginAuthHandler): %v", authSource.Name, err)
+		gothic.BeginAuthHandler(response, request)
+		return nil, goth.User{}, nil
```

Comment:
That's really a hacky patch and I do not think it is right.

Reply:
This is from the official example https://github.com/markbates/goth/blob/master/examples/main.go#L250-L258

Reply:
Yup, it is official, but it is not right.

Reply:
> I have tested it many times manually, and it works.

How did you test? What if your "test" is wrong.